tests/scanners/test.base.js

```diff
@@ -41,82 +41,67 @@ describe('Base Scanner Class', () => {
     expect(baseScannerWithOptions.options.foo).toEqual('bar');
   });
 
-  it('should reject when _getContents is not implemented', () => {
+  it('should reject when _getContents is not implemented', async () => {
     const baseScanner = new BaseScanner('', 'index.html');
 
-    return baseScanner.scan()
-      .then(() => {
-        expect(false).toBe(true);
-      })
-      .catch((err) => {
-        expect(err).toBeInstanceOf(Error);
-        expect(err.message).toEqual('_getContents is not implemented');
-      });
+    await expect(baseScanner.scan()).rejects.toThrow('_getContents is not implemented');
   });
 
-  it('should run all rules', () => {
+  it('should run all rules', async () => {
     const baseScanner = new BaseScannerWithContents('', 'index.html');
 
     const fakeRules = {
       iAmAFakeRule: sinon.stub(),
       iAmAAnotherFakeRule: sinon.stub(),
     };
 
-    return baseScanner.scan(fakeRules)
-      .then(() => {
-        expect(fakeRules.iAmAFakeRule.calledOnce).toBeTruthy();
-        expect(fakeRules.iAmAAnotherFakeRule.calledOnce).toBeTruthy();
-      });
+    await baseScanner.scan(fakeRules);
+    expect(fakeRules.iAmAFakeRule.calledOnce).toBeTruthy();
+    expect(fakeRules.iAmAAnotherFakeRule.calledOnce).toBeTruthy();
```

Comment:
@rpl I thought that sinon assertion was used here just because Sinon was initially used with Mocha and after transition to Jest it wasn't changed. I thought sinon.assert.calledOnce(...) and expect(...).toBeTruthy() do the same, just different syntax for SInon and Jest. Could you please explain the difference?

Reply:
iirc the discussion we had with @EnTeQuAk about it, sinon is still used (and there is a sinon customized configuration in the setup file), there is currently only one place where we are using the Jest mocking feature, because we are mocking a require which is not something that sinon supports).

About the differences between `sinon.assert.calledOnce(...)`, and `expect(...).toBeTruthy()`,
they are mostly the same assertion, but they differ in the error message reported when there is a failure, e.g we could verify these difference by dropping this small test file in the "tests/" dir of the addons-linter repo:

```js
const sinon = require('sinon');

test("test spy with sinon assertions", async () => {
  const aSpy = sinon.spy();
  sinon.assert.calledOnce(aSpy);
});

test("test spy with jest assertions", async () => {
  const aSpy = sinon.spy();
  expect(aSpy.calledOnce).toBeTruthy(); 
});
```

And then execute this single test file using:

```
$ ./node_modules/.bin/jest tests/test.test-asserts-on-sinon-spies.js
``` 
And these are the error messages produced:

```
  ● test spy with sinon assertions

    AssertError: expected spy to be called once but was called 0 times

      4 |   const aSpy = sinon.spy();
      5 | 
    > 6 |   sinon.assert.calledOnce(aSpy);
      7 | });
      8 | 
      9 | test("test spy with jest assertions", async () => {
      
      ...
  ● test spy with jest assertions

    expect(received).toBeTruthy()
    
    Expected value to be truthy, instead received
      false

      10 |   const aSpy = sinon.spy();
      11 | 
    > 12 |   expect(aSpy.calledOnce).toBeTruthy(); 
      13 | });
      14 | 
      
      ...
```

Even in this simple case, the sinon assertion is able to provide an additional information which can be helpful to be able to determine the reason behind a failure on the CI, e.g. in this case the sinon assertion also provide in the error message the number of times that the spy has been actually called with (e.g. it could be 0 by also 2 or more times).

Reply:
@rpl thanks a lot for taking time to make this thorough explanation. It helped a lot. I've made all the changes in files.